integration_tests/src/main/python/join_test.py

```diff
@@ -785,3 +785,37 @@ def do_join(spark):
         return spark.sql("select a.* from {} a, {} b where a.name=b.name".format(
             resultdf_name, resultdf_name))
     assert_gpu_and_cpu_are_equal_collect(do_join)
+
+# ExistenceJoin occurs in the context of existential subqueries (which is rewritten to SemiJoin) if
+# there is an additional condition that may qualify left records even though they don't have
+# join partner records from the right.
+#
+# Thus a query is rewritten roughly as a LeftOuter with an additional Boolean column "exists" added.
+# which feeds into a filter "exists OR someOtherPredicate"
+# If the condition is something like an AND, it makes the result a subset of a SemiJoin, and
+# the optimizer won't use ExistenceJoin.
+@ignore_order(local=True)
+@pytest.mark.parametrize(
+    "allowFallback", [
+        pytest.param('true',
+            marks=pytest.mark.allow_non_gpu('SortMergeJoinExec')),
+        pytest.param('false',
+            marks=pytest.mark.xfail(reason="https://github.com/NVIDIA/spark-rapids/issues/589"))
```

Comment:
I find it a bit odd to add a test that tests for both all-on-GPU and not-all-on-GPU, and only one is ever expected to pass.  Tests should cover what is expected, and when a PR later changes the behavior, that PR also changes the test accordingly.  Having a test do it both ways like this is confusing and wastes test time.

Reply:
The not-all-on-GPU test is there to prove that we are executing the right plan. The GPU-only test with the existing constructs already bails on the fact "SortMergeJoinExec" was not on GPU without verifying it had ExistenceJoin as joinType

Reply:
I still don't understand why we would run the test twice.  The test that is verifying we fallback to the CPU can check the SortMergeExec node to ensure it's an existence join.  Otherwise with this kind of precedent, we'd be running all of our tests twice to verify we're getting the right plan before we run it to put it on the GPU.

Not going to block this, since you're actively working on ExistenceJoin. I'm assuming this will soon revert back to just being a single test to make sure it's on the GPU in that PR.